Sources/HummingbirdJobs/JobQueueDriver.swift

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Logging
+
+/// Job queue protocol.
+///
+/// Defines how to push and pop job data off a queue
+public protocol HBJobQueueDriver: AsyncSequence, Sendable where Element == HBQueuedJob<JobID> {
+    associatedtype JobID: CustomStringConvertible & Sendable
+
+    /// Called when JobQueueHandler is initialised with this queue
+    func onInit() async throws
+    /// Push Job onto queue
+    /// - Returns: Identifier of queued job
+    func push(data: Data) async throws -> JobID
```

Comment:
Is there a reason we're using `Data` here, instead of `ByteBuffer`?

Reply:
Just the fact we are moving back and forward between the two types. Should probably change to ByteBuffer.